File: standing-app-ng/src/app/model/player.ts
import { Team } from "./team";

export class Player {
    id: number;
    firstname: string;
    lastname: string;
    birthday: Date;
    teamId: number;
    bats: string;
    throws: string;
    team: Team;

    constructor(
        id: number = 0,
        firstname: string = '',
        lastname: string = '',
        birthday: Date = new Date(),
        teamId: number = 0,
        bats: string = '',
        throws: string = '',
        team: Team = new Team()
    ) {
        this.id = id;
        this.firstname = firstname;
        this.lastname = lastname;
        this.birthday = birthday;
        this.teamId = teamId;
        this.bats = bats;
        this.throws = throws;
        this.team = team;
    }
}

File: StandingsApp/StandingsApp/Models/EF/Player.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StandingsApp.Models.EF
{
    [Table("Player")]
    public class Player
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Firstname { get; set; }
        
        [Required]
        public string Lastname { get; set; }

        [Required]
        public DateTime Birthday { get; set; }

        [Required]
        public int TeamId { get; set; }

        public string Bats { get; set; }

        public string Throws { get; set; }
    }
}
